File: CSharp/StringCalculator.Logic.Tests/StringCalculator1Test.cs
using System;
using NUnit.Framework;

namespace StringCalculator.Logic.Tests
{
    public class StringCalculator1Test
    {
        [Test]
        public void whenMoreThan2NumbersAreUsedThenExceptionIsThrown()
        {
            Assert.Throws(typeof(ArgumentException), () => StringCalculator1.Add("1,2,3"));
        }

        [Test]
        public void when2NumbersAreUsedThenNoExceptionIsThrown()
        {
            Assert.DoesNotThrow(() => StringCalculator1.Add("1,2"));
        }

        [Test]
        public void whenNonNumberIsUsedThenExceptionIsThrown()
        {
            Assert.Throws(typeof(FormatException), () => StringCalculator1.Add("1,x"));
        }
    }
}

File: Java/StringCalculator/src/test/java/StringCalculator1Test.java
import org.junit.Test;

public class StringCalculator1Test {

	@Test(expected = RuntimeException.class)
	public void whenMoreThan2NumbersAreUsedThenExceptionIsThrown() {
		StringCalculator1.add("1,2,3");
	}
	
	@Test
	public void when2NumbersAreUsedThenNoExceptionIsThrown() {
		StringCalculator1.add("1,3");
		
	}
	
	@Test(expected = RuntimeException.class)
	public void whenNonNumberIsUsedThenExceptionIsThrown() {
		StringCalculator1.add("1,x");
	}

}
